javascript.go

```diff
@@ -0,0 +1,94 @@
+package flixkit
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"strings"
+	"text/template"
+	"unicode"
+)
+
+type SimpleParameter struct {
+	Name string
+	Type string
+}
+
+type TemplateData struct {
+    Version     string
+    Parameters []SimpleParameter
+    Title       string
+    Location    string
+    IsScript    bool
+    IsLocalTemplate bool
+}
+
+//go:embed templates/*.tmpl
+var templateFiles embed.FS
+
+func GenerateJavaScript(flix *FlowInteractionTemplate, templateLocation string, isLocal bool) (string, error) {
+    tmpl, err := template.ParseFS(templateFiles, "templates/*.tmpl")
+    if err != nil {
+        fmt.Println("Error executing template:", err)
+        return "", err
+    }
+
+    methodName := TitleToMethodName(flix.Data.Messages.Title.I18N["en-US"])
```

Comment:
Not sure you can guarantee `en-US` is always there or that title is always there

Reply:
I updated the code to check if "en-US" exists. I could range over a map and check if en-US exists if not use another one. I ended up setting a default value for Title since it's used as method name. If description doesn't exist then it will not be added to js doc comment